src/Stratis.Bitcoin/Base/Deployments/Models/ThresholdStateModel.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+
+namespace Stratis.Bitcoin.Base.Deployments.Models
+{
+    /// <summary>
+    /// Class representing the activation states with the count of blocks in each state.
+    /// </summary>
+    public class ThresholdStateModel
+    {
+        public int DeploymentIndex { get; }
+
+        public ThresholdState? StateValue { get; set; }
+
+        public string ThresholdState { get; set; }
+
+        public int Height { get; }
+
+        public int PeriodStartsHeight { get; }
+
+        public int PeriodEndsHeight { get; }
+
+        public int Votes { get; set; }
+
+        public int Threshold { get; }
+
+        public DateTime? TimeStart { get; }
+
+        public DateTime? TimeTimeOut { get; }
+
+        public ThresholdStateModel(int deploymentIndex, int votes, DateTime? timeStart, DateTime? timeTimeOut, int threshold, int height, int periodStartsHeight, int periodEndsHeight, ThresholdState stateValue, string thresholdState)
```

Comment:
You don't really need a ctor and a default ctor here, you can just have `get; set;` on the property and construct the object when you need it, like we do in the other model classes.

Reply:
Replaced with mutable object.